### ISSUE TITLE
[java] false positive for UselessQualifiedThis

### ISSUE THREAD

**Issue by kevin-wayne** [pmd/pmd#1372]
**Affects PMD Version:** 
6.3

**Rule:**
`UselessQualifiedThis`, `UnusedPrivateField`

**Description:**
False positive when instance variable is an array (or multidimensional array) with the same name as the class. While the code below doesn't follow standard Java naming conventions, it doesn't use the `this` keyword, so the `UselessQualifiedThis` rule should not be relevant.

There is also a false positive for `UnusedPrivateField`.

**Code Sample demonstrating the issue:**

```
public class Board {
    private int[] Board;

    public Board(int[] b) {
        for (int i = 0; i < b.length; i++)
            Board[i] = b[i];
    }

    public int size() {
        return Board.length;
    }
}
```

**Running PMD through:** 
CLI

**Comment by oowekyala** [comment]
The parsing for qualified this syntax appears to be very unhelpful. The `.this` is parsed as a primary suffix, which doesn't have a usesThisModifier method or an image, so there's no way to find that it's actually a qualified this without going proceeding by elimination - which is what the rule does, and it fails here. Well at least it's easy to fix (#1427)

**Comment by kevin-wayne** [comment]
Thanks, much appreciated.

**Comment by jsotuyod** [comment]
I created #1433 for the `UnusedPrivateField` false positive, to track it independently
